code/clue.js

```diff
@@ -1,59 +1,326 @@
+  image: "assets/axe.png",
+  weight: 7,
+  id: "axe"
+};
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const bat = {
+  name: "bat",
+  image: "assets/bat.png",
+  weight: 4,
+  id: "bat"
+};
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const wench = {
+  name: "wench",
+  image: "assets/wench.png",
+  weight: 8,
+  id: "wench"
+};
+
+const pistol = {
+  name: "pistol",
+  image: "assets/pistol.png",
+  weight: 3,
+  id: "pistol"
+};
+
+// Array for suspects, weapons, rooms (rooms is not object)
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
-]
+  prPlum,
+  missScarlett,
+  mrsPeacock,
+  colMustard,
+  mrsWhite,
+];
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  bomb,
+  poison,
+  axe,
+  bat,
+  wench,
+  pistol,
+];
+
+const rooms = [
+  "Dining Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio"
+];
+
+// Random selector, takes an array and returns one random item in the array
+
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-const weapons = []
+// Mystery object 
 
-const rooms = []
+const mystery = {
+  killer : null, 
+  weapon : null,  
+  room: null
+};
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
+// Progress bar for loading cards. Bar will oly be shown once clicked. After animation completes, callback function is called for picking
+// killer.
+
+let i = 0;
+const move = (callback) => {
+  document.getElementById('myProgress').style.display = "block";
+  if (i == 0) {
+    i = 1;
+    const elem = document.getElementById("myBar");
+    let width = 1;
+    const id = setInterval(frame, 75);
+    function frame() {
+      if (width > 100) {
+        clearInterval(id);
+        i = 0;
+        document.getElementById('myProgress').style.display = "none";
+      } else if (width == 100){
+        //callback();
+        width++;
+      } 
+      else {
+        width++;
+        elem.style.width = width + "%";
+      }
+      callback();
+    }
+  }
+}
+
+//Function for fading away text when clicking change button or reveal crime.
+
+const remove = () => {
+  document.getElementById("mystery").classList.remove("h1fade");
+  document.getElementById("mystery").innerHTML = "";
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+//Shufflefavourite weapon. Loops through the array of suspects and replaces favourite weapon
+//with new weapon from randomSelector. After that displays text "weapons changed"
+
+const shuffleFavouriteWeapon = (suspect) => {
```

Comment:
`const shuffleFavouriteWeapon = suspect => { ... }`

Reply:
Needs to be solved

Reply:
The problem is that I´m using parenthesis around the parameter? @EmOlsz

Reply:
Exactly, we don't have to use it since it's only one parameter. Parentheses are obligatory only if we pass more than one parameter